### PR TITLE
install resources in a particular order

### PR DESCRIPTION
**Pull request by dprotaso**
This PR orders the resources applied to help mitigate the installation flakes when using `kubectl apply -f istio.yaml`

/assign @nak3 @arturenault

### CODE REVIEW

File: third_party/library.sh [hunk 1]
```diff
@@ -123,11 +123,36 @@ function generate() {
...
 }
 
+function run_yq() {
+  run_go_tool github.com/mikefarah/yq/v4 yq "$@"
+}
+
 function install_yaml() {
-  if grep -q "CustomResourceDefinition" "$1"; then
-    kubectl apply --selector="knative.dev/crd-install=true" -f "$1"
-    kubectl wait --for=condition=Established --all crd
```
Comment (nak3) [inline third_party/library.sh:129]: We will not wait for `Established` condition for CRD anymore. Is it alright?

**Comment by dprotaso** [comment]
hmmm.. I think `run_go_tool` isn't working when the version is in the module path

File: third_party/library.sh (hunk 1 above)
Comment (dprotaso) [reply third_party/library.sh:129]: Yeah - i dropped it - I want to see if it flaked in the presubmit. N `kubectl` invocation is really slow so I figured it wouldn't come up

File: third_party/library.sh [hunk 2]
```diff
@@ -123,11 +123,36 @@ function generate() {
...
+function run_yq() {
+  GO111MODULE=on run_go_tool github.com/mikefarah/yq/v4 yq "$@"
+}
+
 function install_yaml() {
-  if grep -q "CustomResourceDefinition" "$1"; then
-    kubectl apply --selector="knative.dev/crd-install=true" -f "$1"
-    kubectl wait --for=condition=Established --all crd
-  fi
+  local order=(
```
Comment (arturenault) [inline third_party/library.sh:131]: this is cool. feels like there should be a way in kubectl to specify these sorts of orderings... but i guess that wouldn't be declarative

Comment (dprotaso) [reply third_party/library.sh:131]: I think kubectl just passes things along as they appear in `yaml` - in serving we structure our config directory so the resulting yaml is ordered in such a way so it works most of the time.

The `istioctl` output is somewhat ordered but

**Comment by nak3** [comment]
`kubectl apply` creates the resources in the order they appear in the file. So do we want to change the order in `istio.yaml` by this hack?
Then, users also can just run `kubectl apply -f istio.yaml`.

**Comment by dprotaso** [comment]
> kubectl apply creates the resources in the order they appear in the file. So do we want to change the order in istio.yaml by this hack?
Then, users also can just run kubectl apply -f istio.yaml.

Done - we'll do both. I kept the logic in the install script since you could have races for resources in the same file.

**Comment by dprotaso** [comment]
rebased

**Review by nak3 — APPROVED** [review]

**Pull request merged**